File: dotnet/examples/products/v1/DeleteProductInputSample.cs
// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// [START merchantapi_delete_product_input]
using Google.Apis.Auth.OAuth2;
using Google.Shopping.Merchant.Products.V1;
using System;
using static MerchantApi.Authenticator;

namespace MerchantApi
{
    /// <summary>
    /// This class demonstrates how to delete a product input for a given Merchant Center account.
    /// </summary>
    public class DeleteProductInputSample
    {
        // Deletes a product input from a specified data source.
        public void DeleteProductInput(string merchantId, string productId, string dataSource)
        {
            Console.WriteLine("=================================================================");
            Console.WriteLine("Deleting a ProductInput from a data source...");
            Console.WriteLine("=================================================================");

            // Authenticate using either OAuth or a service account.
            ICredential auth = Authenticator.Authenticate(
                MerchantConfig.Load(),
                ProductsServiceClient.DefaultScopes[0]);

            // Create the ProductInputsServiceClient.
            var productInputsServiceClient = new ProductInputsServiceClientBuilder
            {
                Credential = auth
            }.Build();

            // Create the product name to identify the product.
            string name = ProductInputName.FromAccountProductinput(merchantId, productId).ToString();

            // Create the request to delete the product input.
            DeleteProductInputRequest request = new DeleteProductInputRequest
            {
                Name = name,
                DataSource = dataSource
            };

            try
            {
                Console.WriteLine("Sending deleteProductInput request...");
                // The call returns no response on success.
                productInputsServiceClient.DeleteProductInput(request);
                Console.WriteLine(
                    "Delete successful, note that it may take a few minutes for the delete to update in the system. " +
                    "If you make a products.get or products.list request before a few minutes have passed, " +
                    "the old product data may be returned.");
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }

        public static void Main(string[] args)
        {
            MerchantConfig config = MerchantConfig.Load();
            string merchantId = config.MerchantId.Value.ToString();
            // An ID assigned to a product by Google. In the format:
            // channel~contentLanguage~feedLabel~offerId
            string productId = "online~en~US~sku123";

            // The name of the dataSource from which to delete the product.
            // If it is a primary feed, this will delete the product completely.
            // If it's a supplemental feed, it will only delete the product information
            // from that feed, but the product will still be available from the primary feed.
            // Replace {INSERT_DATASOURCE_ID} with your actual data source ID.
            string dataSourceId = "{INSERT_DATASOURCE_ID}";
            string dataSource = $"accounts/{config.MerchantId}/dataSources/{dataSourceId}";

            var sample = new DeleteProductInputSample();
            sample.DeleteProductInput(merchantId, productId, dataSource);
        }
    }
}
// [END merchantapi_delete_product_input]

File: golang/examples/products/v1/delete_product_input_sample.go
// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package v1

// [START merchantapi_delete_product_input]

import (
	"context"
	"fmt"
	"log"

	"cloud.google.com/go/shopping/merchant/products/apiv1/productspb"

	products "cloud.google.com/go/shopping/merchant/products/apiv1"

	"google.golang.org/api/option"
	"github.com/google/merchant-api-samples/go/collection"
	"github.com/google/merchant-api-samples/go/examples/googleauth"
)

// These are all placeholder values. Please fill in your own details.
const accountIDForDelete = "1234567890"

// The ID of the product. In the format of `contentLanguage~feedLabel~offerId`.
const productIDForDelete = "en~label~sku123"

// The data source from which to delete the product.
// Format: `accounts/{account}/dataSources/{dataSource}`
const dataSourceForDelete = "accounts/1234567890/dataSources/1234567890"

type deleteProductInputSample struct{}

func init() {
	if err := collection.Add("products.productinputs.v1.delete_product_input", &deleteProductInputSample{}); err != nil {
		log.Fatalf("could not add example: %v", err)
	}
}

func (s *deleteProductInputSample) Description() string {
	return "This sample demonstrates how to delete a product input"
}

func (s *deleteProductInputSample) Execute() error {
	ctx := context.Background()

	// Authenticates with Google.
	tokenSource, err := googleauth.AuthWithGoogle(ctx)
	if err != nil {
		return fmt.Errorf("failed to authenticate: %w", err)
	}

	// Creates a new product inputs client.
	productInputsClient, err := products.NewProductInputsClient(ctx, option.WithTokenSource(tokenSource))
	if err != nil {
		return fmt.Errorf("could not create product inputs client: %w", err)
	}
	defer productInputsClient.Close()

	// The name of the product to delete.
	// Format: `accounts/{account}/productinputs/{productinput}`
	name := fmt.Sprintf("accounts/%s/productInputs/%s", accountIDForDelete, productIDForDelete)

	// Creates the request to delete the product input.
	req := &productspb.DeleteProductInputRequest{
		Name:       name,
		DataSource: dataSourceForDelete,
	}

	fmt.Println("Sending deleteProductInput request")
	// The DeleteProductInput method doesn't return a response upon success.
	err = productInputsClient.DeleteProductInput(ctx, req)
	if err != nil {
		return fmt.Errorf("could not delete product input: %w", err)
	}

	fmt.Println("Delete successful, note that it may take a few minutes for the delete to update in the system. If you make a products.get or products.list request before a few minutes have passed, the old product data may be returned.")
	return nil
}

// [END merchantapi_delete_product_input]
